exp/api/v1alpha4/azuremanagedmachinepool_types.go

```diff
@@ -56,6 +56,116 @@ type AzureManagedMachinePoolSpec struct {
 	// ProviderIDList is the unique identifier as specified by the cloud provider.
 	// +optional
 	ProviderIDList []string `json:"providerIDList,omitempty"`
+
```

Comment:
we should not be modifying v1alpha4 types. Only v1beta1 types can be updated in the main branch

Reply:
> @meixingdb what do you think about breaking up this big PR into little small PRs, one for each feature that is getting added? this one is very big which makes it difficult to merge quickly. There are also several fields in here that have already been added in the main branch with separate PR which is causing merge conflicts. Examples of smaller PRs to add AKS types: #1669, #1564, #1910

I completely agree